core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/FabricTokenParser.scala

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import spray.json._
+
+class InvalidJwtTokenException(message: String) extends Exception(message)
+class JwtTokenExpiryMissingException(message: String) extends Exception(message)
+class FabricTokenParser(JWToken: String) {
+  val tokens: Array[String] = JWToken.split("\\.")
+  private var parsedToken: JsValue = JsObject.empty
+
+  if (tokens.length == 3) {
+    // Getting the JWT payload which is second member of [header].[payload].[signature]
+    val payload = tokens(1)
+    // Removing whitespace and url safe characters encoded that might have been added to token
+    val sanitizedPayload = payload.replace('-', '+').replace('_', '/').replaceAll("\\.", "").replaceAll("\\s", "")
+    val decodedPayload = java.util.Base64.getDecoder.decode(sanitizedPayload)
+    val decodedJson = new String(decodedPayload)
+    parsedToken = decodedJson.parseJson
+  }
+  else {
+    throw new InvalidJwtTokenException(s"Invalid JWT token. Here is the token = {$JWToken}")
+    println("Invalid JWT token input.")
```

Comment:
this wont ever run

Reply:
Let's keep this here. In case a trigger to this api is done with a parameter passed incorrectly.